bugsnag/src/main/java/com/bugsnag/util/DeamonThreadFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.bugsnag.util;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A version of {@link Executors#defaultThreadFactory()} which returns deamon threads
+ * This is to prevent applications from hanging waiting for the sessions scheduled task
+ * because deamon threads will be terminated on application shutdown
+ */
+public class DeamonThreadFactory implements ThreadFactory {
```

Comment:
The implementation of this class could be simplified by using the decorator pattern. We could create a field for `Executors.defaultThreadFactory()`, then make the thread returned by `threadFactory.newThread` a daemon before returning. That might also make a good candidate for a unit test.

Reply:
changed

Reply:
also added a test